### ISSUE TITLE
Direct transfers should not pay mediation fees

### ISSUE THREAD

**Issue by rakanalh** [raiden-network/raiden#4815]
## Intro

In #4760, a 2% mediation fee gets added to a certain transfer per hop. This is currently causing problems in the scenario where the target is a direct neighbour of the initiator. 

Using internal routing, when no mediation takes place and a transfer is direct between X & Y, there should be no mediation fees added to the transfer.

**Comment by LefterisJP** [comment]
> In #4760, a 2% mediation fee gets added to a certain transfer per hop

No the mediation fee is for the entire route since internal routing can't know the number of hops beforehand.

> Using internal routing, when no mediation takes place and a transfer is direct between X & Y, there should be no mediation fees added to the transfer.

Yep! Also the PR for this should add a test that there are no fees involved in direct transfers.

**Comment by palango** [comment]
> No the mediation fee percentage is for the entire route since internal routing does not have a good educated guess for the number of hops.

I think this is wrong. The code does know how many hops are in the path.

**Comment by LefterisJP** [comment]
> I think this is wrong. The code does know how many hops are in the path.

Yes it does. My bad, wanted to say that it does not have the latest view of the hops as the PFS does.

**Comment by rakanalh** [comment]
@LefterisJP question:

https://github.com/raiden-network/raiden/blob/f8084c4fe3218a72f38e1152af43c9729575dd75/raiden/routing.py#L213-L215

I can see that the `estimated_fees` are for the whole route... does this make the comment above the calculation invalid?

**Comment by LefterisJP** [comment]
> I can see that the estimated_fees are for the whole route... does this make the comment above the calculation invalid?

Yes the comment is wrong in this case. Please change it.
